src/uct/ib/mlx5/gdaki/gdaki.c

```diff
@@ -378,10 +337,73 @@ uct_rc_gdaki_create_cq(uct_ib_iface_t *ib_iface, uct_ib_dir_t dir,
 ucs_status_t
 uct_rc_gdaki_ep_get_device_ep(uct_ep_h tl_ep, uct_device_ep_h *device_ep_p)
 {
-    uct_rc_gdaki_ep_t *ep = ucs_derived_of(tl_ep, uct_rc_gdaki_ep_t);
+    uct_rc_gdaki_ep_t *ep        = ucs_derived_of(tl_ep, uct_rc_gdaki_ep_t);
+    uct_rc_gdaki_iface_t *iface  = ucs_derived_of(ep->super.super.iface,
+                                                  uct_rc_gdaki_iface_t);
+    uct_rc_gdaki_dev_ep_t dev_ep = {};
+    unsigned cq_size, cqe_size, max_tx;
+    size_t cq_umem_len, cq_umem_offset, qp_umem_offset, dev_ep_size;
+    ucs_status_t status;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxPushCurrent(iface->cuda_ctx));
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    cq_size        = UCS_BIT(ep->cq.cq_length_log);
+    cqe_size       = UCS_BIT(ep->cq.cqe_size_log);
+    cq_umem_len    = cqe_size * cq_size;
+    cq_umem_offset = ucs_align_up_pow2(sizeof(uct_rc_gdaki_dev_ep_t),
+                                       ucs_get_page_size());
+    max_tx         = ep->qp.bb_max;
+    qp_umem_offset = ucs_align_up_pow2(cq_umem_offset + cq_umem_len,
+                                       ucs_get_page_size());
+    dev_ep_size    = qp_umem_offset + (max_tx * MLX5_SEND_WQE_BB);
```

Comment:
Looks like a lot of code duplication... we calculate the same fields in `uct_rc_gdaki_ep_t` init function.
If possible, maybe use some helper func because we don't want to cache it.